kaldb/src/test/java/com/slack/kaldb/logstore/search/queryparser/KaldbQueryParserTest.java

```diff
@@ -15,4 +41,221 @@ public void testInit() {
         .isThrownBy(
             () -> new KaldbQueryParser("test", new StandardAnalyzer(), new ConcurrentHashMap<>()));
   }
+
+  @Test
+  public void testExistsQuery() {
+    // indexed=true analyzed=false storeDocValues=true - Use ReservedField
+    withStringField(LogMessage.ReservedField.SERVICE_NAME.fieldName);
+    // indexed=true analyzed=false storeDocValues=true - Not ReservedField
+    withStringField("my_service_name");
+
+    // indexed=true analyzed=true - Use ReservedField
+    withTextField(LogMessage.ReservedField.USERNAME.fieldName);
+    withTextField(LogMessage.ReservedField.MESSAGE.fieldName);
+    // All texty fields that are not reserved will use analyzed=false aka string field which we
+    // already tested
+
+    // indexed=true storeDocValues=true - Use ReservedField
+    withLongField(LogMessage.ReservedField.DURATION_MS.fieldName);
```

Comment:
I would consider how we can make these tests use less method calls, and encapsulate as much of the logic within the actual tests - even at the expense of duplicating code. When the asserts are directly in the test method, it makes troubleshooting failing tests much quicker in the future.